pkgs/os-specific/linux/msi-ec/default.nix

```
@@ -0,0 +1,51 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  linuxPackages,
+  kernel ? linuxPackages.kernel,
+}:
+stdenv.mkDerivation {
+  pname = "msi-ec-kmods";
+  version = "0-unstable-2024-09-19";
+
+  src = fetchFromGitHub {
+    owner = "m1dugh";
+    repo = "msi-ec";
+    rev = "60aad95e2c8948140232f4c4d70f4ceba821d5a5";
+    hash = "sha256-qgM4KeGoLvbVsii5YR3GTGi2MVlU+JeArzEoA+6A3a0=";
+  };
+
+  dontMakeSourcesWritable = false;
+
+  patchPhase = ''
```

```suggestion
  postPatch = ''
```
Overriding entire phases like this is discouraged ([docs](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#phases-sec-stdenv-phases)).